lib/std/dynamic_library.zig

```diff
@@ -338,9 +338,14 @@ pub const WindowsDynLib = struct {
     }
 
     pub fn openW(path_w: [*:0]const u16) !WindowsDynLib {
-        return WindowsDynLib{
+        var offset: usize = 0;
+        if (path_w[0] == '\\' and path_w[1] == '?' and path_w[2] == '?' and path_w[3] == '\\') {
```

Comment:
Isn't there a `startsWith` we can use for this that works with sentineled pointers?

Reply:
Maybe `mem.startsWith(u16, path_w[0..4], &[_]u16{ '\\', '?', '?', '\\' })`, but `startsWith` may try to be too smart and compare the whole 4 bytes at once and voilá, you get an OOB access.

Reply:
There is only `mem.startsWith` that works with slices, but when slicing we do need to do a length check first that is kind of wasted if we only care about the first four bytes. 
The shortciruiting here handles the length check very elegantly, but it could be possible to introduce a `mem.startsWithZ` that generalizes this method.

Reply:
I guess that would be almost identical to `strncmp` from libc